metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/DirectSqlGetPartition.java

```diff
@@ -573,6 +574,10 @@ public int getPartitionCount(
                     null, null,
                     null, sort, pageable, forceDisableAudit);
             }
+        } catch (QueryTimeoutException e) {
+            // On query timeouts, propagate the error.
```

Comment:
How long is the timeout? and what will Metacat return back to the client side for indicating this is a timeout?

Reply:
It is configurable. Like today, it will return a HTTP 500 error.

Reply:
Just from the PR, it's hard for me to trace how the exception is processed through the server to the client. My suggestion is 
- add logging here to indicate the time out happens with table information
- ensure the timeout message is passed to the client so that they can decide either reduce the number of getPartitions or retry

Reply:
Will add comment